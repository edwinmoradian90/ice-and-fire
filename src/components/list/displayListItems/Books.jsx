import React from 'react'
import { Container, Text } from '../../../utils/styledComponents/main'

export default function Books(props) {
    const {
        name,
        authors,
        numberOfPages,
        publisher,
        country,
        released
    } = props.items.data || ''

    const timeRemover = (releasedDate) => {
        if (releasedDate) {
            const modifiedDate = releasedDate.split('T')
            return modifiedDate[0]
        }
    }

    return (
        <Container>
            <ul>
                <li className='bookName'>
                    <Text justifyContent='flex-start'>
                        Name: {name}
                    </Text>
                </li>
                <li className='author'>
                    <Text justifyContent='flex-start'>
                        Author: {authors}
                    </Text>
                </li>
                <li className='numberOfPages'>
                    <Text justifyContent='flex-start'>
                        Number Of Pages: {numberOfPages} pages
                    </Text>
                </li>
                <li className='publisher'>
                    <Text justifyContent='flex-start'>
                        Published by: {publisher}
                    </Text>
                </li>
                <li className='country'>
                    <Text justifyContent='flex-start'>
                        Published at: {country}
                    </Text>
                </li>
                <li className='released'>
                    <Text justifyContent='flex-start'>
                        Release Date: {timeRemover(released)}
                    </Text>
                </li>
            </ul>
        </Container>
    )
}