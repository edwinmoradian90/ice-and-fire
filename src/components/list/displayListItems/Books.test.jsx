import React from 'react';
import renderer from 'react-test-renderer';
import Books from './Books';

describe('Books', () => {
    const items = {
        data: {
            name: 'test',
            authors: [
                'someone'
            ],
            numberOfPages: '9000',
            publisher: 'someone',
            country: 'somewhere',
            released: 'sometime',
        },
    };
    it('renderers correctly', () => {
        const component = renderer.create(
            <Books items={items} />
        );

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});