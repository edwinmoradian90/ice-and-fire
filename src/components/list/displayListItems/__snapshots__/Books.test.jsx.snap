// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Books renderers correctly 1`] = `
<div
  className="sc-AxirZ jRTNxu"
>
  <ul>
    <li
      className="bookName"
    >
      <div
        className="sc-AxhUy yLbuM"
      >
        Name: 
        test
      </div>
    </li>
    <li
      className="author"
    >
      <div
        className="sc-AxhUy yLbuM"
      >
        Author: 
        someone
      </div>
    </li>
    <li
      className="numberOfPages"
    >
      <div
        className="sc-AxhUy yLbuM"
      >
        Number Of Pages: 
        9000
         pages
      </div>
    </li>
    <li
      className="publisher"
    >
      <div
        className="sc-AxhUy yLbuM"
      >
        Published by: 
        someone
      </div>
    </li>
    <li
      className="country"
    >
      <div
        className="sc-AxhUy yLbuM"
      >
        Published at: 
        somewhere
      </div>
    </li>
    <li
      className="released"
    >
      <div
        className="sc-AxhUy yLbuM"
      >
        Release Date: 
        sometime
      </div>
    </li>
  </ul>
</div>
`;
